File: Level2/221208/JavaScript/Camouflage.js
export default function solution(clothes) {
  const clothsObj = {};
  clothes.forEach((cloth) => (clothsObj[cloth[1]]
    ? clothsObj[cloth[1]]
     += 1 : (clothsObj[cloth[1]] = 1)));

  let answer = 1;
  for (const key of Object.keys(clothsObj)) {
    answer *= clothsObj[key] + 1;
  }

  return answer - 1;
}

File: Level2/230108/Java/Camouflage.java
import java.util.HashMap;

public class Camouflage {
    public int solution(String[][] clothes) {
        int answer = 1;

        HashMap<String, Integer> hashMap = new HashMap<>();

        for(String[] clothe : clothes) {
            hashMap.put(clothe[1], hashMap.getOrDefault(clothe[1], 0) + 1);
        }

        for(String key : hashMap.keySet()) {
            answer *= hashMap.get(key) + 1;
        }

        return answer - 1;
    }
}
